core/src/tile/tileManager.cpp

```diff
@@ -3,237 +3,350 @@
+                logMsg("[%d, %d, %d] Load\n", id.z, id.x, id.y);
+
+                if (!source->loadTileData(task, m_dataCallback)) {
+                    logMsg("ERROR: Loading failed for tile [%d, %d, %d]\n", id.z, id.x, id.y);
+                }
+            }
+            if (m_loadPending == MAX_PARALLEL_DOWNLOADS) {
+                break;
             }
         }
     }
+
+    // logMsg("all:%d loading:%d processing:%d pending:%d\n",
+    //        m_tileSet.size(), m_loadQueue.size(),
+    //        m_queuedTiles.size(), m_loadPending);
+    m_loadQueue.clear();
 }
 
 void TileManager::addTile(const TileID& _tileID) {
-    
+    logMsg("[%d, %d, %d] Add\n", _tileID.z, _tileID.x, _tileID.y);
+
     std::shared_ptr<MapTile> tile(new MapTile(_tileID, m_view->getMapProjection()));
-    m_tileSet[_tileID] = std::move(tile);
 
-    for (auto& source : m_dataSources) {
-        
-        if (!source->loadTileData(_tileID, *this)) {
-            
-            logMsg("ERROR: Loading failed for tile [%d, %d, %d]\n", _tileID.z, _tileID.x, _tileID.y);
-            
-        }
-    }
-    
     //Add Proxy if corresponding proxy MapTile ready
-    updateProxyTiles(_tileID);
+    updateProxyTiles(*tile);
+
+    m_loadQueue.push_back(_tileID);
+    m_tileSet[_tileID] = std::move(tile);
 }
 
 void TileManager::removeTile(std::map< TileID, std::shared_ptr<MapTile> >::iterator& _tileIter) {
-    
     const TileID& id = _tileIter->first;
+    auto& tile = _tileIter->second;
 
-    // Make sure to cancel the network request associated with this tile, then if already fetched remove it from the proocessing queue and the worker managing this tile, if applicable
-    for(auto& dataSource : m_dataSources) {
-        dataSource->cancelLoadingTile(id);
-        cleanProxyTiles(id);
-    }
+    logMsg("[%d, %d, %d] Remove\n", id.z, id.x, id.y);
 
-    // Remove tile from queue, if present
-    const auto& found = std::find_if(m_queuedTiles.begin(), m_queuedTiles.end(), 
-                                        [&](std::unique_ptr<TileTask>& p) {
-                                            return (p->tileID == id);
-                                        });
+    if (setTileState(*tile, MapTile::Canceled)) {
 
-    if (found != m_queuedTiles.end()) {
-        m_queuedTiles.erase(found);
-        cleanProxyTiles(id);
-    }
-    
-    // If a worker is processing this tile, abort it
-    for (const auto& worker : m_workers) {
-        if (!worker->isFree() && worker->getTileID() == id) {
-            worker->abort();
-            // Proxy tiles will be cleaned in update loop
+        // 1. Remove from Datasource
+        // Make sure to cancel the network request associated with this tile.
+        for(auto& dataSource : m_dataSources) {
+            dataSource->cancelLoadingTile(id);
         }
+
+        //// Canceled tiles will be ignored ////
+        // 2. Remove from tiles queued for processing
```

Comment:
Is this a Todo? Can the comments be cleaned here?

Some old code is sitting here (commented out)